### PR TITLE
Fix access for root user's STS credential

### PR DESCRIPTION
**Pull request by donatello**
## Description

Fix a bug in STS credential for root user

## How to test this PR?

Console login as root user should work as expected.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression (https://github.com/minio/minio/commit/890f43ffa5d75b88211cce61e0bb15cd48753040)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: cmd/sts-handlers_test.go [hunk 1]
```diff
@@ -154,6 +155,68 @@ func (s *TestSuiteIAM) TestSTS(c *check) {
...
+		Secure: s.secure,
+	})
+	if err != nil {
+		c.Fatalf("Err creating user admin client: %v", err)
+	}
+	userAdmClient.SetCustomTransport(s.TestSuiteCommon.client.Transport)
+
+	_, err = userAdmClient.ServerInfo(ctx)
+	if err != nil {
+		c.Fatalf("root user STS should be able to get server info")
```
Comment (harshavardhana) [inline cmd/sts-handlers_test.go]:
```suggestion
		c.Fatalf("root user STS should be able to get server info: %v", err)
```

File: cmd/sts-handlers_test.go [hunk 2]
```diff
@@ -154,6 +155,68 @@ func (s *TestSuiteIAM) TestSTS(c *check) {
...
+	userAdmClient, err := madmin.NewWithOptions(s.endpoint, &madmin.Options{
+		Creds:  cr.NewStaticV4(value.AccessKeyID, value.SecretAccessKey, value.SessionToken),
+		Secure: s.secure,
+	})
+	if err != nil {
+		c.Fatalf("Err creating user admin client: %v", err)
+	}
+	userAdmClient.SetCustomTransport(s.TestSuiteCommon.client.Transport)
+
+	_, err = userAdmClient.ServerInfo(ctx)
```
Comment (harshavardhana) [inline cmd/sts-handlers_test.go]:
```suggestion
	_, err = userAdmClient.AccountInfo(ctx)
```

Better to call this API as it returns the list of buckets this user has access, you can validate the returned value is consistent i.e it returns access to all buckets for 'root' user.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/sts-handlers_test.go (hunk 2 above)
Comment (donatello) [reply cmd/sts-handlers_test.go]: ah ok, will update the PR

**Comment by donatello** [comment]
PTAL @harshavardhana

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#4fe3d2b9-2d40-478a-8890-f16575ac0e26) |
<a name="4fe3d2b9-2d40-478a-8890-f16575ac0e26"></a>
### 13947-8744b36/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s3:30956
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 51 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 24 seconds
(8/15) Running minio-dotnet tests ... done in 52 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 12 seconds
(10/15) Running minio-java tests ... done in 31 seconds
(11/15) Running minio-js tests ... FAILED in 54 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook for \"removeObject(bucketName, objectName, removeOpts)",
  "args": "bucketName:minio-js-test-ver-presign657face7-2a4b-40a3-b222-97fb73a423ed ",
  "duration": 5,
  "alert": "datafile-1-b ",
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty. You must delete all versions in the bucket. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:89:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:164:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1334:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(11/15) Running minio-py tests ... done in 1 minutes and 13 seconds
(12/15) Running s3cmd tests ... done in 25 seconds
(13/15) Running s3select tests ... done in 3 seconds
(14/15) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Pull request merged**